### PR TITLE
Add from_receiver method to FormulaEngine

### PR DESCRIPTION
**Pull request by matthias-wende-frequenz**

### CODE REVIEW

**Comment by Marenz** [comment]
CI failing

File: src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py [hunk 1]
```diff
@@ -374,6 +374,24 @@ def new_receiver(
         recv._engine_reference = self  # type: ignore # pylint: disable=protected-access
         return recv
 
+    @staticmethod
+    def from_receiver(receiver: Receiver, name: str) -> FormulaEngine:
+        """
+        Create a formula engine from a receiver.
```
Comment (llucax) [inline src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py]:
```suggestion
        """Create a formula engine from a receiver.
```

File: src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py [hunk 2]
```diff
@@ -374,6 +374,24 @@ def new_receiver(
...
+
+        Args:
+            receiver: A receiver that streams `Sample`s.
+            name: A name for the formula engine.
+
+        Returns:
+            A formula engine that streams the `Sample`s from the receiver.
+        """
+        builder = FormulaBuilder(name)
+        builder.push_metric(name, receiver, nones_are_zeros=False)
```
Comment (llucax) [inline src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py]: Shouldn't `nones_are_zeros` be an option to this method?

**Review by llucax — COMMENTED** [review]
It would be nice to have a test for this, and to write what motivated this in the PR description or commit message.

File: src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py (hunk 2 above)
Comment (matthias-wende-frequenz) [reply src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py]: You are right!

**Comment by matthias-wende-frequenz** [comment]
updated and rebased onto v0.x.x. ~Only tests are still missing.~

**Review by shsms — APPROVED** [review]
A longer term alternative for this would be to bring back specialized formula receivers,  and allow for them to be composed with engines directly, to reduce boiler plate.

**Review by llucax — COMMENTED** [review]
Release notes! @matthias-wende-frequenz

**Pull request merged**